### DISCUSSION TITLE
Opposite of `x`? ie how to shrink selection?

### DISCUSSION THREAD

**Discussion by James-Firth** [helix-editor/helix#9755] [Q&A] [answered]
I was mashing `x` to select a bunch of lines to delete but hit one too many.

I searched for "shrink" in the keymap since I had a feeling that was the right word, but those didn't seem to do it (`Alt-x`). Is there a way to shrink my selection if I go to far without redoing it?

Helix has been great for quickly manipulating data, this case came up when I C&P'd a markdown table but it had a bunch of extra rows I wanted to delete.

I also didn't find anything searching discussions

**Comment by pascalkuthe** [comment, marked as answer]
use selecet modes for that. `x` is just a shortcut for the common case of extending one line down. Mor complex selection manipulations is done by entering select mode and finsihgin with `X`

**Reply by James-Firth** [reply]
Thanks @pascalkuthe ! That makes a lot of sense, I wanted to make sure I wasn't missing a hotkey.

For anyone reading this in the future:
One easy way for me was to mash `x` then hit `v` then `up arrow` then `d` to fix the selection then `escape` to head back to Normal mode.

Alternative just switch to Select Mode with `v` and handle the entire selection manipulation there before deleting

**Comment by chtenb** [comment]
I shrink selections by using the commands select_line_below and select_line_above, which I find easier than what is available in the default key mapping.

https://github.com/helix-editor/helix/pull/9080#issuecomment-2008964831

**Comment by jeffs** [comment]
`vj`

(then `<esc>` if you're not about to do something that puts you back in Normal mode anyway)
